common/src/main/java/org/apache/celeborn/common/network/ssl/SSLFactory.java

```diff
@@ -85,6 +96,28 @@ private void initJdkSslContext(final Builder b) throws IOException, GeneralSecur
     this.jdkSslContext = createSSLContext(requestedProtocol, keyManagers, trustManagers);
   }
 
+  /*
+   * As b.trustStore is null, credulousTrustStoreManagers will be used - and so all
+   * certs will be accepted - and hence self-signed cert from lifecycle manager will
+   * be accepted at executors: nothing else needs to be done to enable its
+   * use - other than generation of the certificate itself, which is what gets done here.
+   */
+  private void configureAutoSsl(Builder b) {
+    // The keystore creation ideally be done only at Lifecycle manager, and not in executors
+    // It does not cause issues to run it at executors, though it is not necessary.
+    // How can be identify this scenario and ensure this is not invoked at executors ?
```

Comment:
Any suggestions on how to accomplish this ?
Not doing it is fine - just slightly better if we can limit this only to driver/job manager

Reply:
Interesting idea !
My concern would be if this makes it slightly difficult to configure ... rpc client module would be the client side to lifecycle manager module at driver, right ? So becomes important to keep some of the configs in sync ... we could inherit the configs, but not sure if that becomes intuitive as project evolves.

Reply:
Oh wait, maybe I misunderstood the idea.
Is the proposal to:
* Use `RPC_LIFECYCLEMANAGER_MODULE` and `RPC_CLIENT_MODULE` internally to differentiate.
* Have both of inherit from `RPC_CLIENT_MODULE` - and users configure for `RPC_CLIENT_MODULE`, not at the internal module level.

This way we have the ability to differentiate internally, without exposing this detail to users ?
I will prototype assuming this is the idea - but appears to be fine to me; will update.

Reply:
Yeah, this is the idea, though I didn't think it in detail... I've merged this PR, let's send a new one :)